src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -485,11 +485,17 @@ bool Unit::IsWithinMeleeRange(Unit const* obj) const
     float dz = GetPositionZMinusOffset() - obj->GetPositionZMinusOffset();
     float distsq = dx*dx + dy*dy + dz*dz;
 
-    float maxdist = GetCombatReach() + obj->GetCombatReach() + 4.0f / 3.0f;
+    float maxdist = GetMeleeRange(obj);
 
     return distsq <= maxdist * maxdist;
 }
 
+float Unit::GetMeleeRange(Unit const* target) const
+{
+    float range = this->GetCombatReach() + target->GetCombatReach() + 4.0f / 3.0f;
```

Comment:
Imo target should be checked before calling GetMeleeRange, cuz using caster as target if !target -- may be not what it should be in some cases.